crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -430,4 +474,160 @@ mod test {
+
+        let packet = FungibleTokenPacketData {
+            denom: "nootasset".to_string(),
+            sender: String::new(),
+            amount: "100".to_string(),
+            receiver: "1c0c490f1b5528d8173c5de46d131160e4b2c0c3".to_string(),
+            memo: String::new(),
+        };
+        let packet_bytes = serde_json::to_vec(&packet).expect("failed to serialize packet data");
+
+        execute_ics20_transfer(
+            &mut state_tx,
+            &packet_bytes,
+            &"source_port".to_string().parse().unwrap(),
+            &"source_channel".to_string().parse().unwrap(),
+            &"dest_port".to_string().parse().unwrap(),
+            &"dest_channel".to_string().parse().unwrap(),
+            false,
+        )
+        .await
+        .unwrap();
+
+        let recipient = Address::try_from_slice(
+            &hex::decode("1c0c490f1b5528d8173c5de46d131160e4b2c0c3").unwrap(),
+        )
+        .unwrap();
+        let denom: Denom = format!("dest_port/dest_channel/{}", "nootasset").into();
+        let balance = state_tx
+            .get_account_balance(recipient, denom.id())
+            .await
+            .unwrap();
+        assert_eq!(balance, 100);
+    }
+
+    #[tokio::test]
+    async fn execute_ics20_transfer_to_bridge_account_ok() {
+        let storage = cnidarium::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut state_tx = StateDelta::new(snapshot.clone());
+
+        let bridge_address = Address::from([99; 20]);
+        let rollup_id = RollupId::from_unhashed_bytes(b"testchainid");
+        let denom: Denom = "nootasset".to_string().into();
+
+        state_tx.put_bridge_account_rollup_id(&bridge_address, &rollup_id);
+        state_tx
+            .put_bridge_account_asset_ids(&bridge_address, &[denom.id()])
+            .unwrap();
+
+        let packet = FungibleTokenPacketData {
+            denom: "nootasset".to_string(),
+            sender: String::new(),
+            amount: "100".to_string(),
+            receiver: hex::encode(bridge_address),
+            memo: "destinationaddress".to_string(),
+        };
+        let packet_bytes = serde_json::to_vec(&packet).expect("failed to serialize packet data");
+
+        execute_ics20_transfer(
+            &mut state_tx,
+            &packet_bytes,
+            &"source_port".to_string().parse().unwrap(),
+            &"source_channel".to_string().parse().unwrap(),
+            &"dest_port".to_string().parse().unwrap(),
+            &"dest_channel".to_string().parse().unwrap(),
+            false,
+        )
+        .await
+        .unwrap();
+
+        let denom: Denom = format!("dest_port/dest_channel/{}", "nootasset").into();
+        let balance = state_tx
+            .get_account_balance(bridge_address, denom.id())
+            .await
+            .unwrap();
```

Comment:
```suggestion
            .expect("there should be a balance because the ics20 transfer moved funds of the given denomination to that address);
```
This would help a future maintainer understand what the intent of this check is.